### PR TITLE
feat: barSeries.startValue added

### LINKED ISSUE #15463: 柱状图是否支持从最小值开始画，现在柱状图都只能从0开始画
### Version
5.1.2

### Reproduction link
[https://segmentfault.com/q/1010000015342610](https://segmentfault.com/q/1010000015342610)

### Steps to reproduce
柱状图是否支持从最小值开始画，现在柱状图都只能从0开始画。

### What is expected?
希望柱状图从最小值开始画，如值为-10，最小值为-20，则希望从-20画到-10，而不是现在的0到-10

### What is actually happening?
希望柱状图从最小值开始画，如值为-10，最小值为-20，则希望从-20画到-10，而不是现在的0到-10

### LINKED ISSUE #16840: [Bug] When polar coordinates are negative, it still renders the bar from 0.
### Version

5.3.0

### Link to Minimal Reproduction

https://codepen.io/billyu811/pen/QWaQZvg

### Steps to Reproduce

1.I want to create a polar chart as usual, but it has negative values ​​on the radius axis.
2.I got data from BE use axios.
3.I didn't get the effect I wanted,I want it to fill the bar towards positive numbers starting from a minimum value.
But it turned out to be this:
https://codepen.io/billyu811/pen/QWaQZvg
4.then I tried to change the min, max, scale, axisLine.onZero of the radius axis, they are not work.
I can't find a solution in the doc.
Until I saw a PR that seemed to fixed the same problem.But I don't know why it reappeared.
https://github.com/apache/echarts/pull/12418

### Current Behavior

I want to create a polar chart, but I got the opposite bar.

### Expected Behavior

I want fill the bar towards positive numbers starting from a minimum value.

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

Please let me know if I'm using a wrong option, or something is wrong with my code, or if it's just a bug.

### LINKED ISSUE #17077: [Bug]the bar chart starts at 0, and setting xAxis.axisline.onZero: false won't make it start at the minimum
### Version

5.3.2

### Link to Minimal Reproduction

_No response_

### Steps to Reproduce

option = {
      xAxis: {
        type: 'category',
        data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'],
        axisLine: {
          onZero: false
        }
      },
      yAxis: {
        type: 'value'
      },
      series: [{
        data: [-120, 200, 150, 80, 70, 110, 130],
        type: 'bar'
      }]
    }

### Current Behavior

In 5.2.2, the bar chart starts at 0, and setting xAxis.axisline.onZero: false won't make it start at the minimum
image

### Expected Behavior

This is OK for versions below 4.0.0

image

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by jiawulin001**
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

barSeries.startValue is added for users to specify where the bars start.

### Fixed issues

- fix #15463
- fix #16840 
- fix #17077

### Before: What was the problem?

As stated in the issues. Users cannot specify the start value of bars.

|issue #16840| issue #17077|
|--|--|
|image|image|

### After: How is it fixed in this PR?

Users now can specify `barSeries.startValue` to set the starting value of bars. Stack and multi-axes are also tested to be running correctly.

|issue #16840 (startValue = -200)| issue #17077 (startValue = -150)|
|--|--|
|after #16840|after #17077|

## Misc

- [x] The API has been changed (apache/echarts-doc#xxx).
- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

bar-startValue.html

### Merging options

- [x] Please squash the commits into a single one when merging.

### CODE REVIEW

**Comment by susiwen8** [comment]
This might fix #15463

**Comment by jiawulin001** [comment]
Sure, it's done.

**Comment by Ovilia** [comment]
This looks to be a useful feature, only that more test cases are required. For example, to test the cases when min or max values are defined for the axis, or when `r` does start from 0 for polar bars.

And you need to run the visual tests to test the differences caused by this PR.

```bash
npm run test:visual
```

Choose the latest nightly build to compare with the one build from your source code.

**Comment by Ovilia** [comment]
It seems there are a lot of commits that are already in the `master` branch. Try [this solution](https://stackoverflow.com/questions/16306012/github-pull-request-showing-commits-that-are-already-in-target-branch#answer-45286781).

**Comment by Ovilia** [comment]
Please also add a `startValue` test with dataZoom component.

**Review by Ovilia — CHANGES_REQUESTED** [review]

**Comment by jiawulin001** [comment]
@Olivia Good intuition on `dataZoom`! I tested and found `startValue` not compatible with `dataZoom`.
I've revised the code and now it works well with `dataZoom` and more naturally incorporate in original code. Also `min/max` set by user now has higher priority over `startValue`. Another change is that `startValue` is moved under `axis` rather than `series`, which I think is more reasonable.
Relevant test cases concerning `dataZoom`, priority and polar are all added. Please check them out.

**Comment by dailijian** [comment]
@jiawulin001 Merging is blocked，

**Review by Ovilia — APPROVED** [review]

File: src/chart/bar/BarSeries.ts [hunk 1]
```diff
@@ -81,6 +81,8 @@ export interface BarSeriesOption
 
     showBackground?: boolean
 
+    startValue?: number
```
Comment (plainheart) [inline src/chart/bar/BarSeries.ts:84]: This option is unused here and **should be removed** in the next version.

**Pull request merged**